client/src/App.js

```diff
@@ -12,26 +13,25 @@ function App() {
     alert(error);
   };
 
-  const getRandomColor = () => {
-    var letters = "0123456789ABCDEF";
-    var color = "#";
-    for (var i = 0; i < 6; i++) {
-      color += letters[Math.floor(Math.random() * 16)];
-    }
-    return color;
-  };
-
+  /* Updates the all the data for the main graph */
   const updateAllData = (data) => {
+    // name for the most recently collected data
     const name = data ? data[0].name : null;
-
     setRowData(data);
 
-    if (name && !(name in allData)) {
-      setAllData((allData) =>
-        !allData.find((data) => data.name === name)
-          ? [...allData, { name: name, color: getRandomColor(), data: data }]
-          : allData
-      );
+    if (name) {
```

Comment:
this is just preventing duplicate data right? can we do this check in the axios call instead? i.e. don't send request to the backend for duplicate data.

Reply:
oh that was just making sure the name is not empty or null. A bit useless since all the data has valid names so I can remove this all together if you want. We can do the check in the axios call too and that simplifies all the conditional logic in this method

Reply:
yep that sounds good.